sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleViewModel.kt

```diff
@@ -119,6 +120,25 @@ abstract class ScheduleViewModel(private var scheduleDao: ScheduledActivityEntit
                 }))
     }
 
+    /**
+     * This should only be called with results of old research stack results
+     * @param schedule to make the metadata info json archive, if null, none will be included
+     * @param taskResult to upload to S3
+     */
+    fun uploadResearchStackTaskResultToS3(schedule: ScheduledActivityEntity?,
```

Comment:
The ViewModel lifecycle may end before the subscription is complete. It'd be safer to have the ScheduleRepository subscribe (or have both ScheduleRepository and this ViewModel subscribe, calling `.cache()` after uploadManager.queueUpload()` so queueUpload only gets called once).

Reply:
That's a good point!